server/mediaserver.go

```diff
@@ -208,6 +224,8 @@ func (s *LivepeerServer) startBroadcast(cpl core.PlaylistManager) (*BroadcastSes
 		OrchestratorInfo: tinfo,
 		OrchestratorOS:   orchOS,
 		BroadcasterOS:    cpl.GetOSSession(),
+		Sender:           s.LivepeerNode.Sender,
+		PMSessionID:      sessionID,
```

Comment:
The intent was to to keep the `pm` package unaware of how it would eventually be used and for as much of the PM related logic to be handled out of the box by the `pm` package. We initially considered storing a `pm.Sender` per `BroadcastSession`, however, we realized we would still eventually need some way to synchronize all these `pm.Sender` instances so that they all have an up to date view of the remaining account deposit so that no `pm.Sender` instance overspends because another instance sent out many winning tickets. So, rather than having the `pm` package user handling deposit view synchronization we can use a single `pm.Sender` instance across each `BroadcastSession`. In this case, we still need a way the single `pm.Sender` instance to keep track of the current `senderNonce` for a set of ticket params, so we use an internal session within the `pm.Sender` that is referenced by its sessionID (which is just `recipientRandHash`).

I realize that the design probably could be simplified since the user of the `pm` package already has a notion of session, but we were interested in keeping the `pm` package as reusable as possible without any assumption that the user already has a notion of a session. The current design of the `pm` package would allow users to use it to send one-to-many out of the box without the user having to implement its own notion of a session. Definitely interested to hear if you have any thoughts on how this design could be improved though!

Reply:
Ah I see. The synchronization could still happen under the hood within `pm` as tickets are created. How will the account synchronization happen? I'm assuming via `TicketBroker` -- which probably needs to be threadsafe anyway.

I'm not sure what's meant by the "user having its own notion of a session". The user needs to store a session ID (or many); it could also store a PM session (or many) that's defined as part of the `pm` package. Storing sessions eliminates that duplicate layer of bookkeeping within the `Sender` and makes it less explicitly stateful. Either way, there is some tracking required from the user.

Also, by storing sessions proper, we don't need to retain session information indefinitely (do we?) as sessions would be GC'd once the connection terminates. Right now, we accumulate sessions in memory without cleaning them up, which is not ideal for a long-running service process. Eventually this will exhaust the available memory.

The one reason I can see the current setup being useful is if the `Sender` needs a way to scan over all the (active) sessions. But I don't really see a clear need for this at the moment.

Perhaps it'd help me to understand how the fully-formed `Sender` interface is intended to work? The comments still indicate there are some TODOs.

To accommodate the current setup, I was thinking something like:

```
interface PMSession {
  CreateTicket(...) // syncs deposit view under the hood via broker?
}
func NewPMSession(broker TicketBroker, params TicketParams) PMSession
...
struct BroadcastSession {
  PMSession session
  ...
}
```

Reply:
> How will the account synchronization happen?

One possibility is for `Sender` to maintain a running view of the account's deposit (which might be different from the on-chain deposit) if the recipient notifies the sender of winning tickets prior to redemption. This would also be compatible with your suggestion of creating a `PMSession` object per each set of ticket params - the `PMSession` object could hold a reference to the `Sender` object and would query its running view of the account's deposit when creating a ticket.

> I'm not sure what's meant by the "user having its own notion of a session". 

I meant that in our case there is already a `BroadcastSession` object used for streams so one approach would be to eliminate the notion of a PM session and simply add additional fields to the `BroadcastSession` object to track PM related information. However, it would be nice for users of the package that do not have the equivalent of a `BroadcastSession` object to not have to worry about defining their own session objects unless they absolutely need to.

> Also, by storing sessions proper, we don't need to retain session information indefinitely (do we?) as sessions would be GC'd once the connection terminates. Right now, we accumulate sessions in memory without cleaning them up, which is not ideal for a long-running service process. Eventually this will exhaust the available memory.

Great point - I really like this benefit of your suggestion.

> Perhaps it'd help me to understand how the fully-formed Sender interface is intended to work?

The original intent was for the global `Sender` object to handle all the state related to sending tickets including tracking internal sessions and `BroadcastSession` would just use `sessionID` to tell the `Sender` object which session it cares about without it knowing anything about the details of the session. Later on, we could add support for receiving new ticket params mid stream and we could replace the existing sessionID in `BroadcastSession` with the new sessionID when this happens.

Overall, I like your suggestion - what do you think about tackling it as a part of the refactor period post-MVS in the interest of not increasing the scope of this PR beyond its core intent (which is just integrating the PM sender into the node and handling RPC messages)?

Reply:
When would a user not have the a `BroadcastSession` (or the equivalent, though not sure what that refers to)?